### PR TITLE
Java: add XStreamMarshaller unsafe deserialization

### PR DESCRIPTION
**Pull request by haby0**
The `spring-oxm` component encapsulates the `XStreamMarshaller` serialization and deserialization objects, and the bottom layer calls `XStream`, which has a deserialization vulnerability.

### CODE REVIEW

File: java/ql/src/semmle/code/java/frameworks/spring/SpringOxm.qll [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import java
```
Comment (smowton) [inline java/ql/src/semmle/code/java/frameworks/spring/SpringOxm.qll:1]:
```suggestion
/** Definitions related to the Spring Object/XML Marshalling library. */

import java
```

File: java/change-notes/2021-05-31-add-xstreammarshaller-unsafe-deserialization.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+lgtm,codescanning
+* The "Deserialization of user-controlled data" (`java/unsafe-deserialization`) query
+  now recognizes `XStreamMarshaller` deserialization.
```
Comment (smowton) [inline java/change-notes/2021-05-31-add-xstreammarshaller-unsafe-deserialization.md:3]:
```suggestion
  now recognizes Spring OXM's `XStreamMarshaller`.
```

**Comment by haby0** [comment]
@smowton Hello, I am not planning to apply for a bonus for this pr, do I still need the intervention of the safety laboratory?

**Comment by smowton** [comment]
No, I indeed hadn't noticed this PR exists for that reason, apologies. I've started an eval to see what this finds; in the meantime please rebase to resolve conflicts against concurrent improvements to the same query.

**Comment by smowton** [comment]
No interesting results (just a bunch of duplicates of a match against the Spring framework itself, noting that depending on the unmarshaller configured it's possible a returning HTTP response could be unsafely unmarshalled).

Therefore for now I won't merge this, as it imposes extra cost and possibly future FPs for no current benefit.

**Comment by haby0** [comment]
OK

**Pull request closed without merging**